### PR TITLE
Add support for hipFFT with PPPMKokkos

### PR DESCRIPTION
**Pull request by emilyviolet**
**Summary**
Added support for AMD's hipFFT library for PPPM when using Kokkos with the HIP backend. Compiling and linking behaves similarly to cuFFT on NVIDIA:
* When compiling with CMake, it will attempt to link with hipFFT if KISSFFT has not been requested.
* When compiling with traditional make, the link commands must be specified manually.
HipFFT is a GPU-only library and cannot be used for CPU calculations.

I've also updated some of the documentation on compiling Kokkos to add information for HIP and hipFFT, but was unsure how much to add. Happy to write extra stuff (e.g. an example `Makefile`) if that would be useful.

**Author(s)**
Emily Kahl, The University of Queensland: e.kahl@uq.edu.au

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**
N/A

**Implementation Notes**
I have tested that the patch compiles on AMD architecture with both CMake and classic Make build systems. The hipFFT build runs on AMD MI100 GPUs for `rhodo` benchmark and other test-cases with long-range electrostatics. I cannot say whether it passes unit tests, as there do not seem to be tests for Kokkos with the GPU backend (only for Kokkos with OpenMP).

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**
N/A

### CODE REVIEW

File: src/KOKKOS/fft3d_kokkos.cpp [hunk 1]
```diff
@@ -63,6 +67,11 @@ FFT3dKokkos<DeviceType>::FFT3dKokkos(LAMMPS *lmp, MPI_Comm comm, int nfast, int
     cudaDeviceGetLimit(&stack_size,cudaLimitStackSize);
     if (stack_size < 2048)
       cudaDeviceSetLimit(cudaLimitStackSize,2048);
+//  #elif defined (KOKKOS_ENABLE_HIP)
```
Comment (akohlmey) [inline src/KOKKOS/fft3d_kokkos.cpp:70]: Please remove.

Comment (emilyviolet) [reply src/KOKKOS/fft3d_kokkos.cpp]: Done

**Comment by stanmoore1** [comment]
Thanks @emilyviolet, will take a look

**Comment by stanmoore1** [comment]
Code looks good, will test on an AMD GPU.

**Review by stanmoore1 — APPROVED** [review]

**Comment by stanmoore1** [comment]
I see correct numerics and a nice speedup with this on an AMD GPU:

KISS:
```
Loop time of 1.47747 on 1 procs for 100 steps with 32000 atoms
Performance: 11.696 ns/day, 2.052 hours/ns, 67.683 timesteps/s
FFT time (% of Kspce) = 0.487835 (74.51)
FFT Gflps 3d (1d only) = 1.0653156 1.2222232
```
hipFFT:
```
Loop time of 1.007 on 1 procs for 100 steps with 32000 atoms
Performance: 17.160 ns/day, 1.399 hours/ns, 99.305 timesteps/s
FFT time (% of Kspce) = 0.0670499 (31.39)
FFT Gflps 3d (1d only) = 7.7509106 69.997966
```

**Comment by stanmoore1** [comment]
@akohlmey this PR is ready to merge from my POV.

**Pull request merged**